backend/src/federation/index.ts

```diff
@@ -4,11 +4,24 @@
 import DHT from '@hyperswarm/dht'
 // import { Connection } from '@dbTools/typeorm'
 import { backendLogger as logger } from '@/server/logger'
+import CONFIG from '@/config'
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const sodium = require('sodium-native')
 
 function between(min: number, max: number) {
   return Math.floor(Math.random() * (max - min + 1) + min)
 }
 
+const getSeed = (): Buffer | null => {
+  if (CONFIG.FEDERATE_DHT_SEED) {
+    const secret = CONFIG.FEDERATE_DHT_SEED
+    const seed = sodium.sodium_malloc(sodium.crypto_box_SEEDBYTES)
```

Comment:
what we need sodium for?
Just check the config for the required length, dont include sodium to have a definition of `32`.

You can just use `Buffer.alloc(CONFIG.FEDERATE_DHT_SEED, ' hex' )`

Reply:
https://github.com/hyperswarm/dht/blob/4190b7505c365ef8a6ad607fc3862780c65eb482/test/connections.js#L554

This one allocs a 32 byte buffer and writes stuff into it - this might be the smartest way?

Reply:
https://github.com/hyperswarm/dht/blob/43b789fd4ed9f12f077350a411bf6cc5c79d6df7/lib/crypto.js#L10

The library does not expose any constant itself. It also just uses plaintext definition of 32 - even tho not ideal I would consider to also do this instead of including sodium

Reply:
if you use `hex` the key must be twice the length: https://www.geeksforgeeks.org/java-program-to-convert-hex-string-to-byte-array/

Two hex chars become 1 byte. 16^2  = 256
You can also choose not to use hex but just plain ascii

Reply:
i will find a solution without sodium...